src/services/sync/repo/user.repo.js

```diff
@@ -0,0 +1,106 @@
+const accountModel = require('../../account/model');
+const { userModel } = require('../../user/model');
+const roleModel = require('../../role/model');
+const { equal: equalIds } = require('../../../helper/compare').ObjectId;
+const { BadRequest } = require('../../../errors');
+
+const createAccount = async (account) => {
+	return accountModel.create({
+		userId: account.userId,
+		username: account.username.toLowerCase(),
+		systemId: account.systemId,
+		activated: true,
+	});
+};
+
+const resolveUserRoles = async (roles) => {
+	return roleModel
+		.find({
+			name: {
+				$in: roles,
+			},
+		})
+		.lean()
+		.exec();
+};
+
+const createUser = async (user) => {
+	user.roles = await resolveUserRoles(user.roles);
+	return userModel.create({
+		firstName: user.firstName,
+		lastName: user.lastName,
+		schoolId: user.schoolId,
+		email: user.email,
+		ldapDn: user.ldapDn,
+		ldapId: user.ldapId,
+		roles: user.roles,
+	});
+};
+
+/**
+ * request user and compare the email address.
+ * if possible it should be solved via unique index on database
+ * @param {string} email
+ * @param {string} userId
+ */
+const checkMail = async (email, userId) => {
+	if (email) {
+		const users = await userModel
+			.find({ query: { email: email.toLowerCase() } })
+			.lean()
+			.exec();
+		if (userId && users.length === 1 && equalIds(users[0]._id, userId)) return;
+		if (users.length !== 0) {
+			throw new BadRequest('Email already exists.');
+		}
+	}
+};
+
+const createUserAndAccount = async (inputUser, inputAccount) => {
```

Comment:
as a personal preference, I think users of this function shouldnt need to know the distinction between inputUser and inputAccount.
I would prefer if this would take a single object, containing both the user informations like email, as well as the account information like username.